### PR TITLE
Replace nose with pytest

### LINKED ISSUE #638: WIP: Remove dependency on nose
I am a maintainer of Python packages in openSUSE, and I am on my crusade of eliminating nose1 from our distribution. When I look at its repository on https://github.com/nose-devs/nose, the last release 1.3.7 was on 2 Jun 2015, and even the last commit on the master branch was on 4 Mar 2016. Moreover, it seems that nose1 will stop working with upcoming Python 3.9 (or 3.10).

[This patch](https://github.com/eventlet/eventlet/files/5070080/remove_nose.patch.txt) eliminates dependency on nose, which has been unnecessary for a long time. Resulting test suite depends on pytest, which is currently perfectly developed and maintained.

Unfortunately, the resulting testsuite (with tarball 0.26.1) doesn’t pass for me: there are 61 on collection of the tests by pytest (which use as the test runner, without explicit dependency on it), see [this complete detailed build log](https://github.com/eventlet/eventlet/files/5070117/_log.txt) for more.

Do you have any idea what I do wrong? Is there a way how to make the test suite to pass even in the environment without the network connection (no Linux distro build machines have network access)? Would it be possible to [mark](https://docs.pytest.org/en/latest/mark.html) some tests as requiring network, so they can be *en mass* skipped?

Thank you for any help.

### PR DESCRIPTION
**Pull request by itamarst**
Fixes #638 

Unless nose3 starts being maintained again, this is a prerequisite for Python 3.12 support. nose3 is, in any case, essentially a dead project.

* [x] Everything passes in CI
* [x] Everything _runs_ in CI. nose runs 722 tests, pytest now runs 722 tests too.

### CODE REVIEW

**Comment by 4383** [comment]
I'm ok to move from nose to pytest, motivations (#638) looks legit.

**Comment by itamarst** [comment]
This is fun. nose when running tests does 722 tests, when you run `nosetests --collect-only` it finds 723 :sob:

File: setup.py [hunk 1]
```diff
@@ -31,7 +31,6 @@
             'README.rst'
         )
     ).read(),
-    test_suite='nose.collector',
```
Comment (itamarst) [inline setup.py:34]: This was for `setup.py test`, which is now deprecated.

**Comment by 4383** [comment]
> This is fun. nose when running tests does 722 tests, when you run `nosetests --collect-only` it finds 723 

I'm just guessing... Maybe 'SkipTest' is just skipped at runtime or something like that, although it seems unlikely that we'll skip only a single test...

File: setup.py (hunk 1 above)
Comment (4383) [reply setup.py:34]: yes, no problem

**Review by 4383 — APPROVED** [review]
God job! Thanks @itamarst

**Pull request merged**